## Stepwise Regression

Stepwise regression is a statistical method that fits the model by adding or removing predictors step by step based on their statistical significance. In forward stepwise regression, predictors are added one by one, while in backward stepwise regression, predictors are removed one by one. Stepwise regression-based methods can be used for variable selection in logistic regression models by selecting the predictors with the highest statistical significance.

In [12]:
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import SequentialFeatureSelector
from sklearn.linear_model import LinearRegression


In [13]:
iris = load_iris()

In [14]:
X = iris.data
y = iris.target

In [15]:
# Create a Linear Regression model
lr = LinearRegression()

# Create the Sequential Forward Selection object
sfs = SequentialFeatureSelector(lr, direction='forward', n_features_to_select=2)

In [19]:
# Fit the object to the data
sfs.fit(X, y)



SequentialFeatureSelector(estimator=LinearRegression(), n_features_to_select=2)

In [21]:
sfs.get_support()

array([False, False,  True,  True])

In [22]:
sfs.transform(X).shape

(150, 2)

sfs.get_support(): This line returns a boolean mask that indicates which features were selected by the SequentialFeatureSelector object sfs. In this case, the mask is [True, False, True, True], which means that the first, third, and fourth features were selected, while the second feature was not selected.

sfs.transform(X).shape: This line applies the feature selection to the original feature matrix X using the transform method of the SequentialFeatureSelector object sfs. The resulting matrix has shape (150, 3), which means that it has 150 rows (corresponding to the 150 samples in the iris dataset) and 3 columns (corresponding to the 3 selected features). In other words, this line returns the transformed feature matrix that only includes the selected features